### ISSUE TITLE
AKS deployments using ARM conflict with Kubernetes Load Balancer when using a static IP address

### ISSUE THREAD

**Issue by bnawijn** [Azure/AKS#1359]
**What is the goal**:
We want to deploy AKS using ARM and have the same static IP address for ingress and egress on the cluster. 

**What happened**:
We were unable to use the same IP address for ingress and egress.

**How to reproduce it (as minimally and precisely as possible)**:
We have a static IP address deployed in it's own resource group within the subscription where we want AKS to be deployed. This IP address must survive the lifetime of the cluster.

We deploy AKS using an ARM template where we include the loadBalancerProfile containing the static IP address we want to use for the outbound traffic:
```
        "networkProfile": {
          "networkPlugin": "kubenet",
          "loadBalancerSku": "standard",
          "podCidr": "10.244.0.0/16",
          "serviceCidr": "10.0.0.0/16",
          "dnsServiceIP": "10.0.0.10",
          "dockerBridgeCidr": "172.17.0.1/16",
          "loadBalancerProfile": {
            "outboundIPs": {
              "publicIPs": [
                {
                  "id": "[parameters('ipAddressResourceId')]"
                }
              ]
            }
          }
        }
```
During the deployment the Azure Load Balancer is configured having a single Frontend-IP configuration, which we want to use for the inbound and outbound traffic through the LB.

After that we deploy our Kubernetes configuration using `kubectl apply` where we setup the Load Balancer Service, using the same static Public IP address:
```
apiVersion: v1
kind: Service
metadata:
  name: azure-aks-ingress-controller
  annotations:
    service.beta.kubernetes.io/azure-load-balancer-resource-group: ##staticPublicIPResourceGroup## 
spec:
  loadBalancerIP: ##staticPublicIPAddress##
  type: LoadBalancer
  ports:
  - port: 80
```

This will result in the Load Balancer in Kubernetes not having an External-IP (it's in the state "pending"). When inspecting the Load Balancer service (`kubectl descripe service azure-aks-ingress-controller`) we see the following warning: 
```
azure-cloud-provider: network.LoadBalancersClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code="PublicIPReferencedByMultipleIPConfigs" Message="Public ip address /subscriptions/xxx/resourceGroups/yyy/providers/Microsoft.Network/publicIPAddresses/zzz is referenced by multiple ipconfigs in resource /subscriptions/xxx/resourceGroups/yyy/providers/Microsoft.Network/loadBalancers/kubernetes." Details=[]
```

Although shown as a warning the result is that the Azure Load Balancer in the AKS resources resource group (MC_***) won't have the expected load balancing rules (including the health probes) setup. Thus, the system is not working correctly.

The work around to this all is to deploy the AKS resources without specifying an outbound IP address, deploy the Kubernetes load balancer service using the static IP address and change the outbound rule on the load balancer to use the frontend-ip configuration that is created when deploying the Kubernetes load balancer service and finally remove the IP address (and frontend-ip configuration) that was generated during the AKS resources deployment.

**What you expected to happen**:
We expected to see the static Public IP as the "external IP" on the Kubernetes ingress AND see that this IP address is used as the outbound IP on AKS. 
Basically we expect the deployment of the Kubernetes load balancer to reuse the available frontend-ip configuration on the load balancer and attach the required rules to this configuration.

**Comment by xuto2** [comment]
It's an issue in k8s cloud provider that it doesn't reuse an existing front-ipconfig with the desired IP.

**Comment by dominik-lekse** [comment]
Also ran into the issue that multiple Service resources cannot use the same static public ip. 

The cloud-provider-azure sig tracks this in https://github.com/kubernetes-sigs/cloud-provider-azure/issues/267.

My request to the AKS team: Please mention this as a limitation in the documentation page https://docs.microsoft.com/en-us/azure/aks/static-ip. I can image there are quite some people running into this.

**Comment by palma21** [comment]
Tracking upstream in https://github.com/kubernetes-sigs/cloud-provider-azure/issues/267 and in https://github.com/Azure/AKS/issues/1621#issuecomment-664006158.

We will add the note to the docs. Thanks for the feedback.

CC @mlearned @zr-msft @laurenhughes @jluk
